src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -617,17 +617,22 @@ bool hasNonIdentityLayout(ValueRange operands) {
 // Support functions for parallel region.
 //===----------------------------------------------------------------------===//
 
-// Return the outermost loop within [firstDim, lastDim) for which (ub-lb) >
-// minSize. Runtime dimensions are assumed to satisfy the size requirement by
-// definition. If found one, it is parDim and the function returns true.
+// Return the outermost loop within [firstInclusiveDim, lastExclusiveDim) for
+// which (ub-lb) > minSize. Runtime dimensions are assumed to satisfy the size
+// requirement by definition. If found one, it is parDim and the function
+// returns true.
 
 bool findSuitableParallelDimension(llvm::SmallVectorImpl<IndexExpr> &lb,
-    llvm::SmallVectorImpl<IndexExpr> &ub, int64_t firstDim, int64_t lastDim,
-    int64_t &parDim, int64_t minSize) {
-  for (int64_t i = firstDim; i < lastDim; ++i) {
+    llvm::SmallVectorImpl<IndexExpr> &ub, int64_t firstInclusiveDim,
+    int64_t lastExclusiveDim, int64_t &parDim, int64_t minSize) {
+  assert(lb.size() == ub.size() && "expected identical ranks for lb/ub");
+  firstInclusiveDim = firstInclusiveDim > 0 ? firstInclusiveDim : 0;
+  lastExclusiveDim =
+      lastExclusiveDim > (int64_t)lb.size() ? lb.size() : lastExclusiveDim;
```

Comment:
Suggest to change to:
```C
if (firstInclusiveDim <= 0)
  firstInclusiveDim = 0;
if (lastExclusiveDim > (int64_t)lb.size())
  lastExclusiveDim = lb.size()
```

since the value is unchanged when condition is not met.